documentation/my-homeworks/week2/todo-app/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+//import TodoList from './components/todo-list'
+import Container from './components/Container'
+
+
+const data = require('./todo.json');
+
+
+class App extends Component {
+  state = {
+    todoItems: []
+  }
+
+
+  componentWillMount() {
+    localStorage.getItem('todoItems') && this.setState({
+      todoItems: JSON.parse(localStorage.getItem('todoItems'))
+    })
+  }
+
+  componentDidMount() {
+    if (!localStorage.getItem('todoItems')) {
+      this.setState({
+        todoItems: data
+      })
+    } else {
+      console.log('we are using the data from the Local Storage.')
+    }
+
+  }
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem('todoItems', JSON.stringify(nextState.todoItems))
+  }
+
+  handelDoneStatus = (index) => {
+    const todoItems = JSON.parse(localStorage.getItem('todoItems'))
```

Comment:
You should not depend on localStorage at all to get your state, because you already have a state.

First solution
```
const todoItems = this.state.todoItems.map((item, i) => {
      if (i === index) {
        item.done = !item.done;
      }
      return item;
    });
```
Use map to change the required item.

Second solution 
```
const todoItems = [...this.state. todoItems] // clone the array
todoItems[index].done = true
this.setState({todoItems})
```

Third solution

```
const items = this.state.todoItems;

const todoItems = [...items.slice(0, index), 
       {...items[index], done: !items[index]},
        ...items.slice(index + 1)]

this.setState({todoItems});
```